romancal/dq_init/dq_initialization.py

```diff
@@ -0,0 +1,84 @@
+import logging
+import numpy as np
+from roman_datamodels.datamodels import ImageModel
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+# Guide star mode exposure types
+GUIDER_LIST = ['WFI_WIM_ACQ','WFI_WIM_TRACK','WFI_WSM_ACQ1','WFI_WSM_ACQ2','WFI_WSM_TRACK']
+
+def do_dqinit(input_model, mask=None):
+    """Check that the input model pixeldq attribute has the same dimensions as
+    the image plane of the input model science data, call apply_dqinit, and update
+    log and cal_step.
+
+    Parameters
+    ----------
+    input_model : input Roman datamodel
+        The Roman datamodel to be data quality  corrected
+
+    mask_model : Roman data model, or None
+        The mask model to use in the data quality correction
+
+    Returns
+    -------
+    output_model : Roman datamodel
+        The data quality corrected Roman datamodel
+    """
+
+    # Initialize the output model as a copy of the input
+    output_model = input_model.copy()
+
+    # Determine if mask is shapewise compatable with the input
+    skip_step = False
+    if (input_model.meta.exposure.type in GUIDER_LIST) or \
+        (isinstance(input_model,ImageModel)):
```

Comment:
Is the data array in a raw GUIDER type exposure a 2D array and not 3D? Why is this opened with an `ImageModel`?

Reply:
ImageModel removed.